Moved PokeNames to PokeDictionary and added localized Pokemon descriptions

yutamago opened the pull request:
This PR flags PokeNames as deprecated and moves it's functionality to PokeDictionary.
PokeDictionary contains a new method that returns the localized Pokémon descriptions for each Pokémon as seen in the Pokémon GO Pokédex.

**Changes made:**
- Added class PokeDictionary
- Moved functionality of PokeNames to PokeDictionary
- Tagged PokeNames as deprecated
- Changed DisplayPokenamesExample to use PokeDictionary instead of PokeNames
- Added static method PokeDictionary.getDisplayDescription(int pokedexId, Locale locale)
- Added translated pokemon_descriptions_xx.xml for German, English, Spanish, French, Italian, Japanese and Korean

Commit:
Added PokeDescriptions.getDisplayDescription(...)
Added description translations for 7 locales

gionata-bisciari commented:
Could "@Deprecated" tag help? without deleting the existing classes?

Commit:
Added missing Pokemon #710 in English locale

Commit:
Added missing Pokemon #710 in default locale

Commit:
- Made PokeNames deprecated.
- Merged PokeNames and PokeDescriptions into PokeDictionary
- Changed PokenameExample to use PokeDictionary instead of PokeNames

yutamago commented:
@gionata-bisciari Yeah, that should do it. Pushed the new changes.

Commit:
Added throw statement to getResourceBundle(..)

Commit:
Remove unused imports

Commit:
Replaced point with comma for better Javadoc

AbandonedCart commented on library/src/main/java/com/pokegoapi/util/PokeNames.java:
```diff
@@ -19,7 +19,7 @@
 
 /**
- * Deprecated. Use {@link PokeDictionary} instead.
+ * Deprecated, use {@link PokeDictionary} instead.
```
```
* @deprecated use {@link PokeDictionary}
```

Will give you the "pretty" deprecation message
(Ugh, the git markup can be tedious)

yutamago replied:
Thanks. I put this in my next commit.

FabianTerhorst commented:
Does we relay need to deprecate classes or method when we not even on a stable version. Everybody thats updates from 0.4 to 0.5 knows that something had changed. A migration.md file would be nice for that.

AbandonedCart commented:
@FabianTerhorst As you have suggested to others yourself, most are building from the development branch. An official release would be the time to remove the deprecated class, while this allows time to convert before an official release is... released.

Commit:
Add pretty deprecation Javadocs

Commit:
Moved file names to constants and renamed getResourceBundle to getPokeBundle

FabianTerhorst commented on library/src/main/java/com/pokegoapi/util/PokeDictionary.java:
```diff
@@ -0,0 +1,88 @@
+package com.pokegoapi.util;
```
add license header

Commit:
Added license header

Commit:
Corrected checkstyle warnings

FabianTerhorst commented:
you want to remove the folder with the old translations? https://github.com/Grover-c13/PokeGOAPI-Java/tree/Development/classes/production/PokeGOAPI-Java

Commit:
- Moved translation example to new TranslatePokenameExample
- Moved supportedLocales to PokeDictionary
- Deleted duplicate Resource Bundle pokemon_names from \classes\production\PokeGOAPI-Java

dojo1017 commented:
I have changed the Fallback method to return Locale.ENGLISH instead of Locale.getDefault().
That should lessen the exception probability to near zero for Locales that have no pokemon names and/or descriptions.

yutamago commented on library/src/main/java/com/pokegoapi/util/PokeDictionary.java:
```diff
@@ -15,27 +15,51 @@
+		return ResourceBundle.getBundle(bundleBaseName, locale, new ResourceBundle.Control() {
+			@Override
+			public Locale getFallbackLocale(String baseName, Locale locale) {
+				return Locale.ENGLISH;
```
I have overwritten the Fallback method to return Locale.ENGLISH instead of Locale.getDefault().
That should lessen the exception probability to near zero for Locales that have no pokemon names and/or descriptions.

yutamago commented:
Feel free to merge if you think this is ready. I have nothing more to add.
